libs/core/langchain_core/prompts/chat.py

```diff
@@ -626,20 +625,31 @@ def format(self, **kwargs: Any) -> BaseMessage:
         if isinstance(self.prompt, StringPromptTemplate):
             text = self.prompt.format(**kwargs)
             return self._msg_class(
-                content=text, additional_kwargs=self.additional_kwargs
+                content=text, additional_kwargs=self.additional_kwargs, name=self.name
             )
         else:
             content: List = []
             for prompt in self.prompt:
                 inputs = {var: kwargs[var] for var in prompt.input_variables}
                 if isinstance(prompt, StringPromptTemplate):
-                    formatted: Union[str, ImageURL] = prompt.format(**inputs)
+                    formatted: Union[str, dict] = prompt.format(**inputs)
                     content.append({"type": "text", "text": formatted})
                 elif isinstance(prompt, ImagePromptTemplate):
-                    formatted = prompt.format(**inputs)
+                    formatted = cast(dict, prompt.format(**inputs))
```

Comment:
why is this return type not a dict?

Reply:
its a typed dict that mypy is annoying about